### PR TITLE
Fix return type of ast.NodeTransformer.generic_visit

### PR DESCRIPTION
**Pull request by Dentosal**
Fixes issue #2485.

### CODE REVIEW

File: stdlib/3/ast.pyi [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 from _ast import *
 
 class NodeVisitor():
     def visit(self, node: AST) -> Any: ...
-    def generic_visit(self, node: AST) -> None: ...
+    def generic_visit(self, node: AST) -> Any: ...
 
 class NodeTransformer(NodeVisitor):
-    def generic_visit(self, node: AST) -> None: ...
+    def generic_visit(self, node: AST) -> Optional[AST]: ...
```
Comment (srittau) [inline stdlib/3/ast.pyi]: Something I missed while initially looking at this: Why return `Optional`? It seems to be `generic_visit()` always returns the node passed in. The note in https://docs.python.org/3.7/library/ast.html about returning `None` refers to the `visit()` method, I think?

Comment (thautwarm) [reply stdlib/3/ast.pyi]: `Optional` return is okay, as well as `List[AST]`.

See https://github.com/python/cpython/blob/c8bb467f4006fbf5d24d2491248bcbabee5d827e/Lib/ast.py#L310 .

The correct return type is `Union[AST, None, List[AST]]`.

Comment (thautwarm) [reply stdlib/3/ast.pyi]:
```python
value = self.visit(value) 
if value is None:
    continue
elif not isinstance(value, AST):
     new_values.extend(value)
     continue
```
Note that `self.visit` is invoked and the return is `value`, so `if value is None` and `new_values.extend(value)` could indicate the type exactly..

Comment (srittau) [reply stdlib/3/ast.pyi]: Good point, I missed that.

**Pull request merged**